jsonb-generator/src/main/java/io/avaje/jsonb/generator/TypeReader.java

```diff
@@ -262,19 +263,42 @@ private void readMethod(Element element, List<FieldReader> localFields) {
         }
       }
     }
-    // for getter/accessor methods only, not setters
-    PropertyPrism.getOptionalOn(methodElement).ifPresent(propertyPrism -> {
-      if (!methodElement.getParameters().isEmpty()) {
-        logError(errorContext + baseType + ", @Json.Property can only be placed on Getter Methods, but on %s", methodElement);
-        return;
-      }
-
-      // getter property as simulated read-only field with getter method
-      final var frequency = frequency(propertyPrism.value());
-      final var reader = new FieldReader(element, namingConvention, currentSubType, genericTypeParams, frequency);
-      reader.getterMethod(new MethodReader(methodElement));
-      localFields.add(reader);
-    });
+    boolean recordAccessor = isRecordAccessor(methodElement);
+    if (!recordAccessor)
```

Comment:
I don't understand the reason for this change. Why skip reading Property annotation for records?

Reply:
It skips property reading for the _method element_ of a record component.

Essentially, what happened is that with

```java
record Some(@Property("other") String value){}
```

you have both the field and the method element annotated with the `@Property("other")`

Reply:
So to fix we skip reading the method property annotation if it is the built-in record accessor

Reply:
What if you have the annotation on the accessor method?

```java
record Some(String value) {

  @Property("other") 
  String value() {
    return value; // or something interesting
  }
}
```

Reply:
you got me there, give me a sec

Reply:
I'd like to get you out of the habit of doing 2 unrelated changes in 1 PR

Reply:
In my mind, it went into the single category of "fix JDT errors"

Reply:
especially since these issues only surfaced under specific conditions when using eclipse. I only noticed it because the cancel generation pr worked a bit too well.

Reply:
> What if you have the annotation on the accessor method?

looks like it doesn't take effect without my changes either, that's a separate problem to fix

Reply:
I'm ok with the package change.  I'm not ok with this record accessor change - it is not clear what problem you have observed with JDT compiling the record with `@Json.Property` (did it result in some duplicate property or something else etc).

Reply:
see the other PR